data-serving/scripts/export-data/transform.py

```diff
@@ -271,30 +274,54 @@ def open_writers(formats: list[str], fields: list[str], output: str):
         print("Error occurred in open_writers():")
         print(e)
     finally:
+        if output == '-':
+            return
         for fmt in files:
             if fmt == "json":
                 files[fmt].write("]\n")
             files[fmt].close()
 
 
-def parse():
+@contextmanager
+def open_input(input_filename: Optional[str]):
+    try:
+        f = open(input_filename) if input_filename else sys.stdin
+        yield f
+    except Exception as e:
+        print("Error occurred in open_input():")
+        print(e)
+    finally:
+        f.close()
+
+
+def transform(output: str, formats: list[str], input: str = None):
+    with open_input(input) as inputfile:
+        headers, fields = get_fields(inputfile)
+        reader = csv.DictReader(inputfile, fieldnames=headers)
+        hasrows = False
+        with open_writers(formats, fields, output) as writers:
+            for i, row in enumerate(map(convert_row, reader)):
+                hasrows = True
+                writerow(formats, writers, row, i)
+        if output != '-' and not hasrows:  # cleanup empty files
+            for fmt in formats:
+                with suppress(FileNotFoundError):
+                    os.remove(f"{output}.{fmt}.gz")
```

Comment:
Simpler as `if exists then remove`?

Reply:
Path().unlink(missing_ok=True) is even better :)